libraries/tock-tbf/src/types.rs

```diff
@@ -304,6 +346,165 @@ impl core::convert::TryFrom<&[u8]> for TbfHeaderV2FixedAddresses {
     }
 }
 
+impl core::convert::TryFrom<&[u8]> for TbfHeaderDriverPermission {
+    type Error = TbfParseError;
+
+    fn try_from(b: &[u8]) -> Result<TbfHeaderDriverPermission, Self::Error> {
+        Ok(TbfHeaderDriverPermission {
+            driver_number: u32::from_le_bytes(
+                b.get(0..4)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+            offset: u32::from_le_bytes(
+                b.get(4..8)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+            allowed_commands: u64::from_le_bytes(
+                b.get(8..16)
+                    .ok_or(TbfParseError::InternalError)?
+                    .try_into()?,
+            ),
+        })
+    }
+}
+
+impl<const L: usize> core::convert::TryFrom<&[u8]> for TbfHeaderV2Permissions<L> {
+    type Error = TbfParseError;
+
+    fn try_from(b: &[u8]) -> Result<TbfHeaderV2Permissions<L>, Self::Error> {
+        let length = u16::from_le_bytes(
+            b.get(0..2)
+                .ok_or(TbfParseError::BadTlvEntry(
+                    TbfHeaderTypes::TbfHeaderPermissions as usize,
+                ))?
+                .try_into()?,
+        );
+
+        if (length as usize) > L {
+            return Err(TbfParseError::TooManyEntries(
+                TbfHeaderTypes::TbfHeaderPermissions as usize,
+            ));
+        }
+
+        let mut perms: [TbfHeaderDriverPermission; L] = [TbfHeaderDriverPermission {
+            driver_number: 0,
+            offset: 0,
+            allowed_commands: 0,
+        }; L];
+
+        for i in 0..length as usize {
+            let start = 2 + (i * size_of::<TbfHeaderDriverPermission>());
+            let end = start + size_of::<TbfHeaderDriverPermission>();
+            if let Some(perm) = perms.get_mut(i) {
+                *perm = b
+                    .get(start..end as usize)
+                    .ok_or(TbfParseError::BadTlvEntry(
+                        TbfHeaderTypes::TbfHeaderPermissions as usize,
+                    ))?
+                    .try_into()?;
+            } else {
+                return Err(TbfParseError::BadTlvEntry(
+                    TbfHeaderTypes::TbfHeaderPermissions as usize,
```

Comment:
Why is this error different than the one on line 385? Can't we get rid of the check on line 385?